file.go

```diff
@@ -100,3 +97,33 @@ func (f *File) WriteTo(w io.Writer) (int64, error) {
 
 	return buf.WriteTo(w)
 }
+
+func (f *File) writeContent() {
+	f.contentTypesWriter()
+	f.workbookWriter()
+	f.workbookRelsWriter()
+	f.worksheetWriter()
+	f.styleSheetWriter()
+}
+
+// GenerateReader provides a function to generate a Reader reading from the file
+func (f *File) GenerateReader() (io.Reader, error) {
```

Comment:
I suggest that create a new function `GetBuffer()` instead of `GenerateReader()` like:

```go
// WriteToBuffer provides a function to get bytes.Buffer from the saved file.
func (f *File) WriteToBuffer() (*bytes.Buffer, error) {
	buf := new(bytes.Buffer)
	zw := zip.NewWriter(buf)
	f.contentTypesWriter()
	f.workbookWriter()
	f.workbookRelsWriter()
	f.worksheetWriter()
	f.styleSheetWriter()
	for path, content := range f.XLSX {
		fi, err := zw.Create(path)
		if err != nil {
			return buf, err
		}
		_, err = fi.Write(content)
		if err != nil {
			return buf, err
		}
	}
	return buf, zw.Close()
}
```

and the user can get `io.Reader` like this:

```go
buf, _ := f.WriteToBuffer()
strings.NewReader(buf.String())
```